src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -54,22 +57,31 @@ public void RunApiCompat()
                     IAssemblySymbol rightSymbols;
                     using (Stream rightAssemblyStream = GetFileStreamFromPackage(_rightPackagePath, rightTuple.rightAssembly.AssemblyId))
                     {
-                        rightSymbols = new AssemblySymbolLoader().LoadAssembly(rightTuple.rightAssembly.AssemblyName, rightAssemblyStream);
+                        rightSymbols = GetAssemblySymbolFromStream(rightAssemblyStream, rightTuple.rightAssembly, out bool resolvedReferences);
+                        runWithReferences &= resolvedReferences;
                     }
+
                     rightContainerList.Add(new ElementContainer<IAssemblySymbol>(rightSymbols, rightTuple.rightAssembly));
                 }
 
+                _differ.RunningWithReferences = runWithReferences;
                 IEnumerable<(MetadataInformation, MetadataInformation, IEnumerable<CompatDifference>)> differences =
                     _differ.GetDifferences(leftContainer, rightContainerList);
 
                 int counter = 0;
                 foreach ((MetadataInformation, MetadataInformation, IEnumerable<CompatDifference> differences) diff in differences)
                 {
                     (MetadataInformation rightAssembly, string header) rightTuple = _dict[left][counter++];
-                    _log.LogMessage(MessageImportance.Low, rightTuple.header);
 
+                    bool logHeaderMessage = true;
                     foreach (CompatDifference difference in diff.differences)
                     {
+                        if (logHeaderMessage)
```

Comment:
Why are we logging this for all the differences ? wouldnt it be the same value for all ?

Reply:
Not sure what you mean, what this if-check is doing is making sure that we only print the header once for each group of differences (from the rightTuple) which seems correct to me.

Reply:
The header tells which assemblies are being compared and has no information specific to api compat differences.

Reply:
Correct. What I am saying is that we are still only logging the header once per left-right tuple, since when looping through the differences we log it the first time but then set `logHeaderMessage` to false so that we don't log it more than once. The only thing this change is doing, is delaying the time we log the header to when we actually know that diff.differences is not empty, which again, seems correct to me.